src/lib/codec-importer.ts

```diff
@@ -0,0 +1,47 @@
+import multicodecs from 'multicodec'
+import { BlockCodec } from 'multiformats/codecs/interface'
+import type { PBNode } from '@ipld/dag-pb'
+import type { IPLDFormat } from 'blockcodec-to-ipld-format'
+
+export default async function codecImporter<T extends any = unknown>(codecCode: number): Promise<BlockCodec<any, any> | IPLDFormat<any>> {
+  // const codecName: string = multicodecs.codeToName[codecCode as CodecCode]
+  // console.log(`codecName: `, codecName);
+  switch (codecCode) {
+    case multicodecs.DAG_CBOR:
+      return await import('@ipld/dag-cbor') as BlockCodec<typeof multicodecs.DAG_CBOR, T>
+    case multicodecs.DAG_PB:
+      return await import('@ipld/dag-pb') as BlockCodec<typeof multicodecs.DAG_PB, PBNode>
+    case multicodecs.GIT_RAW:
+      // @ts-expect-error - no ipld-git types
+      return (await import('ipld-git')).default as BlockCodec<typeof multicodecs.GIT_RAW, T>
+    case multicodecs.RAW:
+      return await import('multiformats/codecs/raw') as BlockCodec<typeof multicodecs.RAW, Uint8Array>
+    case multicodecs.JSON:
+      return await import('multiformats/codecs/json') as BlockCodec<typeof multicodecs.JSON, T>
+    case multicodecs.ETH_ACCOUNT_SNAPSHOT:
+      // @ts-expect-error - no ipld-ethereum types
+      return await import('ipld-ethereum/eth-account-snapshot') as BlockCodec<typeof multicodecs.ETH_ACCOUNT_SNAPSHOT, T>
+    case multicodecs.ETH_BLOCK:
+      // @ts-ignore - no ipld-ethereum types
+      return (await import('ipld-ethereum/eth-block')).default as IPLDFormat<T>
+    case multicodecs.ETH_BLOCK_LIST:
+      // @ts-ignore - no ipld-ethereum types
+      return (await import('ipld-ethereum/eth-block-list')).default as IPLDFormat<T>
+    case multicodecs.ETH_STATE_TRIE:
+      // @ts-ignore - no ipld-ethereum types
+      return (await import('ipld-ethereum/eth-state-trie')).default as IPLDFormat<T>
+    case multicodecs.ETH_STORAGE_TRIE:
+      // @ts-ignore - no ipld-ethereum types
+      return (await import('ipld-ethereum/eth-storage-trie')).default as IPLDFormat<T>
+    case multicodecs.ETH_TX:
+      // @ts-ignore - no ipld-ethereum types
+      return (await import('ipld-ethereum/eth-tx')).default as IPLDFormat<T>
+    case multicodecs.ETH_TX_TRIE:
+      // @ts-ignore - no ipld-ethereum types
+      return (await import('ipld-ethereum/eth-tx-trie')).default as IPLDFormat<T>
```

Comment:
@lidel we're removing ethereum example (https://github.com/ipld/explore.ipld.io/issues/74#issuecomment-1563485950), but we still want to keep these, right?

Reply:
i'm going to remove the dependency on `blockcodec-to-ipld-format` and require future codec support to be of the BlockCodec interface.

Reply:
@SgtPooki  I honestly dont know a single soul that is using these Ethereum codecs. IF we have tests for this, it does not hurt to have them, ok to keep, but if there is no tests, remove until someone asks for them.

To be extra safe, you may ask folks more familiar with Filecoin / FEVM, they may know if we need this, but afaik It's dead and not maintained(?) https://github.com/ipld/js-ipld-ethereum

Reply:
I don't know the specifics, but generally agreed we should showcase functionality that works and is maintained.  I think we were lucky enough to resuscitate the IPLD Explorer.  It's ok if we had to make some cuts so it doesn't topple over again.

Reply:
Ping @i-norden on this one, through a grant, vulcanize took over dev of these and we have shiny new typescript eth codecs that are compatible with all their work to extract and store ethereum chain data and state in ipld form.

In terms of audience, the reach would be minimal for these but they are a novel addition when viewing through an IPLD lens. Maybe @i-norden would like to suggest whether a block or two, or three, or some minimal DAG, could be pinned and shown through the explorer to show something interesting about eth-as-ipld?

Reply:
Hey all and thanks for the ping @rvagg. As Rod said we maintain the new TS/JS codec implementation [here](https://github.com/vulcanize/ts-dag-eth) it is compatible with js-multiformats. We also maintain the new go-ipld-prime based Go implementation [here](https://github.com/vulcanize/go-codec-dageth) that has a plugin for kubo but is not currently shipped with kubo.

These currently only support all the execution layer types, but we plan on maintaining them indefinitely for our internal uses which will also eventually require adding in beacon chain/consensus layer support.

Vulcanize also has _most_ of the ethereum blockchain indexed in a [go-blockstore-compatible format](https://github.com/cerc-io/ipld-eth-db) but we aren't currently pinning any of that to IPFS. We could potentially provide/pin the blocks necessary for an example though.

For an interesting example that includes some interoperability we could traverse from an ethereum header to a NFT mint transaction and then to the corresponding receipt containing a `Mint` event that emits the minted `tokenID`. Or traverse down from a header and search through all receipts in that block until you find one with a `Mint` event. You could then use that `tokenID` to traverse to the contract storage slots necessary to construct the `tokenURI` which could be a CID linking to some media on IPFS. For most contracts its not really possible to do this with a "pure" traversal though because they store pieces of the URI across multiple storage slots and the only definitive way to know the position of those storage slots is to have a priori knowledge about the precompiled contract code (you need to know when the corresponding contract variables were declared as this determines which storage slots are used for storing their values). 

So it'd be like:

1. Starting with ethereum header CID
* Get and decode ethereum header
* Get transaction trie, rct trie, and state trie root CIDs from header
2. Starting from transaction trie root CID 
* Iterate through the transaction trie until you get to the tx of interest
* Note the index (the tx and rct tries are keyed by transaction index)
3. Starting from the receipt trie root CID
* Directed descent down to the same index in the receipt trie to get the corresponding receipt
* Decode receipt
* Extract contract address from the receipt
* Search through the receipt logs for a `Mint` event and extract `tokenID` from it
4. Starting from the state trie root CID
* Directed descent down to the contract of interest based on address pulled from the receipt
*  **Here's where we need to have some prior knowledge about the contract layout to progress any further** 
* If its a contract that uses CIDs for tokenURIs and we know the slot position for a tokenURI mapping (or whatever variable the contract decided to store the CID in) we can use that slot position + the `tokenID` from the event to form the key we want to descend down to in the trie to access the `tokenURI` CID
* Get and decode the image/media/object referenced by that CID (possibly a DAG JSON object itself with links to image and other metadata fields).

Not really simple and not sure how feasible let alone desirable that would be in this context. It is not standard to emit the `tokenURI` during mints due to the additional gas cost (and because most these contracts have capabilities to change the URIs after the mint 🙄) but we could probably find a non-standard contract that does this and that would simplify it to just the tx and rct trie traversals.

Another, possibly simpler, example that includes interoperability would be to traverse to some receipts with events that emit hashes from other blockchains (e.g. bridge, rollup, sidechain contracts) but to continue down that link would require codec support for the chain on the other side (perhaps could find a btc ordinal tx reference somewhere, or maybe a filecoin object CID??).

For something _much_ simple but without interoperability (but maybe still interesting?) we could traverse from a header down to a user's state account to show their ether balance.